client/signup/steps/domains/index.jsx

```diff
@@ -248,8 +248,20 @@ export class RenderDomainsStep extends Component {
 		const productToRemove = this.props.cart.products.find(
 			( product ) => product.meta === domain_name
 		);
+
+		this.setState( ( prevState ) => ( {
+			domainRemovalQueue: [
+				...prevState.domainRemovalQueue,
+				{ meta: productToRemove.meta, productSlug: productToRemove.product_slug },
+			],
+		} ) );
+
 		if ( productToRemove ) {
 			this.removeDomain( { domain_name, product_slug: productToRemove.product_slug } );
+		} else if ( this.state.temporaryCart?.length > 0 ) {
+			this.setState( ( state ) => ( {
+				temporaryCart: state.temporaryCart.filter( ( domain ) => domain.meta !== domain_name ),
+			} ) );
```

Comment:
Here we're ensuring multiple requests for auto-deletion on failed domains won't fail and it will always auto-delete not available domains.